kernel.go

```diff
@@ -164,6 +171,10 @@ func runKernel(connectionFile string) {
 			}
 		}
 	}
+
+	shutdownHeartbeat()
```

Comment:
Should this be deferred?

Reply:
I think maybe just a little more clarity via comments around the heatbeat and waitgroup would be good.

Reply:
The reason I set up a wait group was because the other channels should also be running on other threads. Especially the control thread who's main purpose is to be a shell channel for higher priority requests that don't get blocked by the shell channel. In the future they should all be using it so I thought it was best to just implement now since some synchronization is needed anyways, even if just for one thread.

The intention is to spawn all the channels and wait for them to close before returning from `runKernel`. Possibly calling all of the shutdowns in the shutdown handler?